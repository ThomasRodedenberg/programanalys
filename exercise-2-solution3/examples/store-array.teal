fun main() = {
    // Storing into arrays.
    var a : array[nonnull int] := new array[nonnull int](2); // N 3 57 =, N 3 4 =
    a[0] := 1; // N 4 12 =, N 4 6 =, N 4 4 +, NI 4 4 +
    a[1] := null; // N 5 12 =, N 5 6 =, N 5 4 0, NI 5 4 +

    // How does load work then?
    var x : nonnull int := a[0]; // N 8 29 =, NI 8 27 +, N 8 4 + 

    var b : array[int] := new array[int](2); // N 10 41 =, N 10 4 =
    b[0] := null; // N 11 12 =, N 11 6 =, N 11 4 =, NI 11 4 +
    // N 0 0 =
}