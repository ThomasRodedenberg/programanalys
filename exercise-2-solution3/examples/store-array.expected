N 4 4 +
maybe: NI 4 4 +
N 5 4 0
maybe: NI 5 4 +
NI 8 27 +
choice: N 8 4 ?  |  N 8 4 +
maybe: NI 11 4 +

